Zend/zend_API.c

```diff
@@ -667,9 +671,13 @@ static const char *zend_parse_arg_impl(int arg_num, zval *arg, va_list *va, cons
 
 				if (!zend_parse_arg_object(arg, p, ce, check_null)) {
 					if (ce) {
-						return ZSTR_VAL(ce->name);
+						if (check_null) {
+							return ZSTR_VAL(ce->name);
```

Comment:
this is a todo. should I use a smart_str?

Reply:
```
zend_spprintf(error, 0, "of type ?%s", ZSTR_VAL(ce->name));
return "";
```
should probably do it?

Reply:
Yes, it will work! I remembered that there was a segfault the last time I did something similar, that's the reason I asked it. But it has just come to my mind that I had already fixed that issue...